src/DecryptionFailureTracker.ts

```diff
@@ -288,40 +324,36 @@ export class DecryptionFailureTracker {
             this.userTrustsOwnIdentity = (
                 await crypto.getUserVerificationStatus(client.getUserId()!)
             ).isCrossSigningVerified();
-            this.keysChangedHandler = () => {
-                this.handleKeysChanged(client).catch((e) => {
-                    console.log("Error handling KeysChanged event", e);
-                });
-            };
-            client.on(CryptoEvent.KeysChanged, this.keysChangedHandler);
         }
-
-        this.baseProperties = baseProperties;
     }
 
-    /**
-     * Start checking for and tracking failures.
-     */
-    public start(): void {
-        this.setClient(MatrixClientPeg.safeGet());
-        this.checkInterval = window.setInterval(
-            () => this.checkFailures(Date.now()),
-            DecryptionFailureTracker.CHECK_INTERVAL_MS,
-        );
-
-        this.trackInterval = window.setInterval(() => this.trackFailures(), DecryptionFailureTracker.TRACK_INTERVAL_MS);
+    private registerHandlers(client: MatrixClient): void {
+        // After the client attempts to decrypt an event, we examine it to see
+        // if it needs to be reported.
+        const decryptedHandler = (e: MatrixEvent): void => this.eventDecrypted(e, Date.now());
+        // When our keys change, we check if the cross-signing keys are now trusted.
+        const keysChangedHandler = (): void => {
+            this.handleKeysChanged(client).catch((e) => {
+                console.log("Error handling KeysChanged event", e);
+            });
+        };
+        // When logging out, remove our handlers and destroy state
+        const loggedOutHandler = (): void => {
+            client.removeListener(MatrixEventEvent.Decrypted, decryptedHandler);
+            client.removeListener(CryptoEvent.KeysChanged, keysChangedHandler);
+            client.removeListener(HttpApiEvent.SessionLoggedOut, loggedOutHandler);
+            self.stop();
+        };
+
+        client.on(MatrixEventEvent.Decrypted, decryptedHandler);
+        client.on(CryptoEvent.KeysChanged, keysChangedHandler);
+        client.on(HttpApiEvent.SessionLoggedOut, loggedOutHandler);
     }
 
     /**
      * Clear state and stop checking for and tracking failures.
      */
     public stop(): void {
-        if (this.client && this.keysChangedHandler) {
-            this.client.removeListener(CryptoEvent.KeysChanged, this.keysChangedHandler);
-        }
-        this.client = undefined;
-        this.keysChangedHandler = undefined;
-
```

Comment:
I guess it's fine in practice because `stop` is never called, but this does mean that `start()` and `stop()` are now asymmetric. If the intention is that `stop` is never called, maybe we should make it private and update the documentation on the class so that it doesn't say "Call ... `stop()` to stop tracking."